specs/EnhancedSecurityMode.md

```diff
@@ -0,0 +1,468 @@
+
+For reference, in the screenshot below, this API sets the levels of ESM as a WebView2 API.
+
+![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/82386753/35977716-e46c-4257-82da-906b0c6f833e)
+
+`CoreWebView2Profile.EnhancedSecurityModeBypassList` - this API allows developers to view and add a URI filter from the ESM bypass list. If a site is in the bypass list, the ESM level for the site will always be set to Off when a user navigates to it, regardless of the value of `CoreWebView2Profile.PreferredEnhancedSecurityModeLevel`. However, this is not respected if ESM is disabled in `CoreWebView2EnvironmentOptions.IsEnhancedSecurityModeEnabled`. That means, when developers set the property when ESM is disabled, it will be updated and persisted, but will not take effect until the feature is enabled.  
+
+`CoreWebView2Profile.EnhancedSecurityModeEnforceList` - this API allows developers to view and add a URI filter from the ESM enforce list. If a site is in the enforce list, the ESM level for the site will always be set to Strict when a user navigates to it, regardless of the value of `CoreWebView2Profile.PreferredEnhancedSecurityModeLevel`. However, similar to the allow list, this is not respected if ESM is disabled in `CoreWebView2EnvironmentOptions.IsEnhancedSecurityModeEnabled`. 
+
+For reference, in the screenshot below, this API allows you to manage the Enforce and Bypass List of ESM as a WebView2API.
+![image](https://github.com/MicrosoftEdge/WebView2Feedback/assets/82386753/98085785-bfae-4de0-bb39-c85e85913e1f)
+
+# Examples
+## IsEnhancedSecurityModeEnabled
+```c#
+Create WebView Environment with option that disable Enhanced Security Mode feature.
+
+void CreateEnvironmentWithOption()
+{
+    CoreWebView2EnvironmentOptions options = new CoreWebView2EnvironmentOptions();
+    // Disable IsEnhancedSecurityModeEnabled by default to reduce impact on performance
+    options.IsEnhancedSecurityModeEnabled = false;
+    CoreWebView2Environment environment = await CoreWebView2Environment.CreateAsync(BrowserExecutableFolder, UserDataFolder, options);
+}
+```
+```cpp
+void AppWindow::InitializeWebView()
+{
+    auto options = Microsoft::WRL::Make<CoreWebView2EnvironmentOptions>();
+    Microsoft::WRL::ComPtr<ICoreWebView2StagingEnvironmentOptions9> optionsStaging9;
+    if (options.As(&optionsStaging9) == S_OK)
+    {
+        CHECK_FAILURE(optionsStaging9->put_IsEnhancedSecurityModeEnabled(
+        m_EnhancedSecurityModeEnabled ? TRUE : FALSE));
```

Comment:
The C# & C++ sample code needs to match in logic. The C# always sets the IsESMEnabled to a constant while this does not.